src/selectors/layout.test.js

```diff
@@ -27,8 +32,51 @@ describe('Selectors', () => {
         extraNodes = extraNodes.concat(extraNodeGroup);
       });
       data.nodes = data.nodes.concat(extraNodes);
-      const customMockState = prepareState({ data });
-      expect(getTriggerLargeGraphWarning(customMockState)).toEqual(true);
+      return data;
+    };
+
+    it('returns false for a small dataset', () => {
+      expect(getTriggerLargeGraphWarning(mockState.animals)).toBe(false);
+    });
+
+    it('returns true for a large dataset', () => {
+      const state = prepareState({ data: prepareLargeDataset() });
+      expect(getTriggerLargeGraphWarning(state)).toBe(true);
+    });
+
+    it('returns false if the sizewarning flag is false', () => {
+      const state = reducer(
+        prepareState({ data: prepareLargeDataset() }),
+        changeFlag('sizewarning', false)
+      );
+      expect(getTriggerLargeGraphWarning(state)).toBe(false);
+    });
+
+    it('returns false if ignoreLargeWarning is true', () => {
+      const state = reducer(
+        prepareState({ data: prepareLargeDataset() }),
+        toggleIgnoreLargeWarning(true)
+      );
+      expect(getTriggerLargeGraphWarning(state)).toBe(false);
+    });
+
+    it('returns false if layout has already been calculated', () => {
```

Comment:
I wonder which scenario this test case refers to? I am not sure whether this case would exist and hence needed given that the graphWarning would be triggered before the layout is calculated ( i.e `updateGraph` is triggered.)

Reply:
This is the case where, instead of clicking 'render anyway', the user enables a filter, then disables it again. This use-case is exactly why we added the `graphHasNodes` argument - so that if the user has already dismissed the warning via a filter, it won't appear again. This is because we only want the warning to appear once on load, and not for subsequent filter changes.

If necessary I can add more detail to explain this in the JSDoc comment for the layout selector?

Reply:
aargh I see - thanks for the clarification! 

It might be clearer to state the case as `'returns false once the user has selected a filter that triggered a layout calculation'` ( or something along those lines) that will indicate the expected behaviour of the warning only appearing once during the lifetime of the session. 
( This could also be further elaborated with this comment: ` // This is the ensure that the warning will only appear once during the lifetime of the session`)

Reply:
Thanks, no worries!

I'm unsure about this part: `‘returns false once the user has selected a filter that triggered a layout calculation’`.
On the one hand, this is more specific so it gives more of an indication of the typical flow, but it’s not the only situation that might cause this behaviour, so I think it's inaccurate. We want the test name to only refer to the actual circumstances that cause the test to pass. The motivation for the feature can be handled by the comment.

I'll add the other comment. Thanks!